apps/reference/docs/guides/platform/logs.mdx

```diff
@@ -4,78 +4,215 @@ title: Logging
+To query for a nested value, add a join for each array level:
 
-#### Timestamp Behavior
+```sql
+select timestamp, request.method, header.cf_ipcountry
+from edge_logs t
+cross join unnest(t.metadata) as metadata
+cross join unnest(metadata.request) as request
+cross join unnest(request.headers) as header
+```
 
-Each log entry is stored with a `timestamp`. In order to utilize the `timestamp` field in a query, you can use the appropriate [timestamp functions](https://cloud.google.com/bigquery/docs/reference/standard-sql/timestamp_functions#timestamp).
+This surfaces the following columns available for selection:
 
-:::note
+<ThemedImage
+  alt="With Two Level Unnesting"
+  sources={{
+    light: useBaseUrl("/img/guides/platform/logs/unnesting-2.png"),
+    dark: useBaseUrl("/img/guides/platform/logs/unnesting-2-dark.png"),
+  }}
+  style={{maxHeight: 200}}
+/>
 
-In the Log Explorer, timestamps are rendered as unix microsecond timestamps. SQL queries, however, should always use the `TIMESTAMP` data type. If you are using a unix timestamp value in a query, cast the value to a `TIMESTAMP` data type.
+This allows you to select the `method` and `cf_ipcountry` columns. In JS dot notation, the full paths for each selected column are:
 
-:::
+- `metadata[].request[].method`
+- `metadata[].request[].headers[].cf_ipcountry`
+
+### LIMIT and Result Row Limitations
+
+The Logs Explorer has a maximum of 1000 rows per run. Use `LIMIT` to optimize your queries by reducing the number of rows returned further.
 
-## Templates
+### Best Practices
 
-Templates are available to help craft you log queries. Templates are available under the **Templates** tab, or under the **Templates** Dropdown in the **Query** tab.
+1. Include a filter over **timestamp**
+
+Querying your entire log history might seem appealing. For **Enterprise** customers that have a large retention range, you run the risk of timeouts due additional time required to scan the larger dataset.
+
+2. Avoid selecting large nested objects. Select individual values instead.
+
+When querying large objects, the columnar storage engine selects each column associated with each nested key, resulting in a large number of columns being selected. This inadvertently impacts the query speed and may result in timeouts or memory errors, especially for projects with a lot of logs.
+
+Instead, select only the values required.
+
+```sql
+-- ❌ Avoid doing this
+select
+  datetime(timestamp),
+  m as metadata  -- <- metadata contains many nested keys
+from edge_logs t
+cross join unnest(t.metadata) as m;
+
+-- ✅ Do this
+select
+datetime(timestamp),
+r.method   -- <- select only the required values
+from edge_logs t
+cross join unnest(t.metadata) as m
+cross join unnest(m.request) as r
+```
```

Comment:
Indent the content within each best practice. Can you also double check the indentation for the 2nd SQL example?
````suggestion
1. Include a filter over **timestamp**.

  Querying your entire log history might seem appealing. For **Enterprise** customers that have a large retention range, you run the risk of timeouts due additional time required to scan the larger dataset.

2. Avoid selecting large nested objects. Select individual values instead.

  When querying large objects, the columnar storage engine selects each column associated with each nested key, resulting in a large number of columns being selected. This inadvertently impacts the query speed and may result in timeouts or memory errors, especially for projects with a lot of logs. Instead, select only the values required.

  ```sql
  -- ❌ Avoid doing this
  select
    datetime(timestamp),
    m as metadata  -- <- metadata contains many nested keys
  from edge_logs t
  cross join unnest(t.metadata) as m;

  -- ✅ Do this
  select
  datetime(timestamp),
  r.method   -- <- select only the required values
  from edge_logs t
  cross join unnest(t.metadata) as m
  cross join unnest(m.request) as r
  ```
````